### ISSUE TITLE
error in FeatureMetricKeypointOptimizerOptions

### ISSUE THREAD

**Issue by zhirui-gao** [Wayne-Mai/traf_public#1]
Hi， thanks for your great work! I count a problem when running the code. Looking forward your reply, thanks!
```
 File "/home/gzr/workspace/NeRF/PixSfM/pixsfm/refine_hloc.py", line 129, in refine_keypoints_and_get_track
    keypoints, ka_data, feature_manager,track_labels,score_labels,root_labels,graph = self.run_ka_and_get_track(
  File "/home/gzr/workspace/NeRF/PixSfM/pixsfm/refine_colmap.py", line 109, in run_ka_and_get_track
    ka_data = self.keypoint_adjuster.refine_multilevel(
  File "/home/gzr/workspace/NeRF/PixSfM/pixsfm/keypoint_adjustment/main.py", line 127, in refine_multilevel
    out = self.refine(keypoints_dict,
  File "/home/gzr/workspace/NeRF/PixSfM/pixsfm/keypoint_adjustment/main.py", line 217, in refine
    solver = ka.FeatureMetricKeypointOptimizer(
TypeError: __init__(): incompatible constructor arguments. The following argument types are supported:
    1. pixsfm._pixsfm._keypoint_adjustment.FeatureMetricKeypointOptimizer(arg0: pixsfm._pixsfm._keypoint_adjustment.FeatureMetricKeypointOptimizerOptions, arg1: pixsfm._pixsfm._keypoint_adjustment.KeypointAdjustmentSetup, arg2: pixsfm._pixsfm._base.InterpolationConfig)

Invoked with: {'loss': {'name': 'cauchy', 'params': [0.25]}, 'solver': {'function_tolerance': 0.0, 'gradient_tolerance': 0.0, 'parameter_tolerance': 1e-05, 'minimizer_progress_to_stdout': False, 'max_num_iterations': 100, 'max_linear_solver_iterations': 200, 'max_num_consecutive_invalid_steps': 10, 'max_consecutive_nonmonotonic_steps': 10, 'use_inner_iterations': False, 'use_nonmonotonic_steps': False, 'update_state_every_iteration': False, 'num_threads': 1, 'callbacks': []}, 'print_summary': False, 'bound': 4.0, 'num_threads': -1, 'root_regularize_weight': -1, 'weight_by_sim': True, 'root_edges_only': False}, <pixsfm._pixsfm._keypoint_adjustment.KeypointAdjustmentSetup object at 0x7ad034e6fd70>, {'nodes': [[0.0, 0.0]], 'mode': 'BICUBIC', 'l2_normalize': True, 'ncc_normalize': False, 'use_float_simd': False}

```

**Comment by zhirui-gao** [comment]
I reinstall ceres, pycolmap and pyceres, the error above go away. But it runs with exit code
```
solver = ka.FeatureMetricKeypointOptimizer(
    to_optim_ctr(self.conf.optimizer, self.callbacks),
    problem_setup,
    to_ctr(self.conf.interpolation))

Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```

**Comment by Wayne-Mai** [comment]
Hi @zhirui-gao , thank you for your interest! I will have a look into it tomorrow. In the meantime, have tried the shared precomputed key points? Is that working?

**Comment by zhirui-gao** [comment]
> Hi @zhirui-gao , thank you for your interest! I will have a look into it tomorrow. In the meantime, have tried the shared precomputed key points? Is that working?

Thanks for your reply! It works when using the shared precomputed key points, but I want to run some other dataset. I think it is a problem with pixsfm. I would like to ask if you have any insights when encountering this problem.

**Comment by Wayne-Mai** [comment]
Sorry to hear that. I managed to set up the environment again but I can't reproduce this issue. 

Yes, it seems like there is a problem with PixSfM. I have tracked your errors and it seems like it's not coming from the part I have modified.

I find similar issues are reported in their repo too, https://github.com/cvg/pixel-perfect-sfm/issues/106 and https://github.com/cvg/pixel-perfect-sfm/issues/131 .  I hope these posts can help.
